### PR TITLE
HW1: Tasks 1 and 2

### PR DESCRIPTION
**Pull request by ArtiFisher**

### CODE REVIEW

File: Task1/1.1/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { Transform } = require('stream');
+
+const reverser = new Transform ({
+  transform: (data, encoding, callback) => {
```
Comment (byslavik) [inline Task1/1.1/index.js:4]: I would suggest to have error handling somewhere here. For example to check if the line is empty

File: Task1/1.1/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    callback(
+      null,
+      data
+      .toString('utf8')
```
Comment (byslavik) [inline Task1/1.1/index.js:8]: What do you think about adding eslint to the project? It helps to maintain your code more readable during all course

File: Task1/1.2/RAMfile.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const csv = require('csvtojson');
+const fs = require('fs');
+
+const filename = 'Task1/1.2/csv/Example'
```
Comment (byslavik) [inline Task1/1.2/RAMfile.js]: you can use `path.resolve(__dirname, 'filename.csv')` to avoid writing path to the file

File: Task1/1.2/RAMfile.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+const input = `${filename}.csv`
+const output = `${filename}.txt`
+
+csv()
+.on('error', console.error)
+.fromFile(input)
+.then(data => {
+    const writer = fs.createWriteStream(output)
```
Comment (byslavik) [inline Task1/1.2/RAMfile.js]: Please check output format of file. There should be some processing of books

File: Task1/1.1/index.js (hunk 2 above)
Comment (ArtiFisher) [reply Task1/1.1/index.js:8]: I am working on it in scope of second task, where I am going to enable typescript.

File: Task1/1.2/RAMfile.js (hunk 4 above)
Comment (ArtiFisher) [reply Task1/1.2/RAMfile.js]: As far as I can see format of output follows example. Does that look different for you?
https://epam.sharepoint.com/sites/EPAMNode.jsGlobalMentoringProgram/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FEPAMNode%2EjsGlobalMentoringProgram%2FShared%20Documents%2FGeneral%2FAssets%2Fnodejs%2Dhw1%2Dex2%2Etxt&parent=%2Fsites%2FEPAMNode%2EjsGlobalMentoringProgram%2FShared%20Documents%2FGeneral%2FAssets&p=true&originalPath=aHR0cHM6Ly9lcGFtLnNoYXJlcG9pbnQuY29tLzp0Oi9zL0VQQU1Ob2RlLmpzR2xvYmFsTWVudG9yaW5nUHJvZ3JhbS9FVnJ1M2hxQ1JYSk9oOTNGXy10NmV3QUJZSV9Cd3V4ZkpfZXBqcWxBYVdySk9BP3J0aW1lPW12MXBBYW8zMkVn

File: Task1/1.2/RAMfile.js (hunk 3 above)
Comment (ArtiFisher) [reply Task1/1.2/RAMfile.js]: Fixed.

File: Task1/1.2/RAMfile.js (hunk 4 above)
Comment (ArtiFisher) [reply Task1/1.2/RAMfile.js]: Fixed format of output for both pipeline and RAM versions.

File: Task1/1.2/formatter.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Transform } from 'stream';
+
+export default new Transform ({
+  transform: (data, encoding, callback) => {
+    const entries = Object.entries(JSON.parse(data.toString('utf8')))
+        .filter(([key]) => key != 'Amount')
+        .map(([key, value]) => [key.toLowerCase(), key === 'Price' ? parseFloat(value) : value]);
```
Comment (byslavik) [inline Task1/1.2/formatter.js:7]: For Price field I would suggest to try to use `colParser` property from configuration object from `csvtojson` package. It is not required, just another way of implementation

Comment (byslavik) [reply Task1/1.2/formatter.js:7]: Also you can read/try to use csv().subscribe method. It also allows you to make some transformations, but a bit in another way

**Comment by byslavik** [comment]
Just 2 small suggestions, but actually looks good to go

File: Task1/1.2/formatter.js (hunk 5 above)
Comment (ArtiFisher) [reply Task1/1.2/formatter.js:7]: ok, will know

**Pull request merged**